### ISSUE TITLE
Crash with go 1.6

### ISSUE THREAD

**Issue by carsso** [root-gg/wigo#2]
Wigo crashes randomly without log with go 1.6.

**One solution we've found is to build wigo with go 1.5.3.**

Deb and Rpm repos packages were affected since Wigo 0.73.8 (builded with go 1.6).
Wigo 0.73.10 packages are fine (builded with go 1.5.3)

**Comment by speed47** [comment]
No log at all ? Got crashes with go1.6 with a stacktrace. Will try go1.5 to see if they go away, but it might as well be a sanity check they added under 1.6 for maps ?

```
fatal error: concurrent map read and map write
goroutine 165 [running]:
runtime.throw(0xb1e460, 0x21)
        /home/speed/go/src/runtime/panic.go:530 +0x90 fp=0xc8204d7b00 sp=0xc8204d7ae8
runtime.mapaccess1_faststr(0x903c00, 0xc8201eb860, 0xc820075e1a, 0xa, 0xc8205164b8)
        /home/speed/go/src/runtime/hashmap_fast.go:202 +0x5b fp=0xc8204d7b60 sp=0xc8204d7b00
wigo.(*Wigo).RecomputeGlobalStatus(0xc8201686e0)
        /home/speed/wigo/src/wigo/global.go:317 +0xd0 fp=0xc8204d7c20 sp=0xc8204d7b60
wigo.(*Host).AddOrUpdateProbe(0xc8201fc6c0, 0xc8204cc240)
        /home/speed/wigo/src/wigo/host.go:69 +0x100 fp=0xc8204d7c60 sp=0xc8204d7c20
main.execProbe(0xc820153d10, 0x26, 0x3b)
        /home/speed/wigo/src/wigo.go:422 +0x1708 fp=0xc8204d7f88 sp=0xc8204d7c60
runtime.goexit()
        /home/speed/go/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc8204d7f90 sp=0xc8204d7f88
created by main.threadLocalChecks.func1.1
        /home/speed/wigo/src/wigo.go:252 +0x1080
```

**Comment by carsso** [comment]
No log in logs files because of the crash.
I've tried to debug it. I've start the process attached and with a strace but it never crashed after a few days.
I've talked about this with @camathieu and he tells me he has the same kind of problem with other projects and we probably need to patch some code.
Rolling back to 1.5 was a quick fix but we need a permanent one for sure.
I've not dug this yet.

**Comment by speed47** [comment]
OK, mine must be another unrelated bug then. Will see if I can reproduce easily or not.

**Comment by speed47** [comment]
... aaaand yep, that's the same bug. New feature of go1.6 [https://golang.org/doc/go1.6](https://golang.org/doc/go1.6)

> The runtime has added lightweight, best-effort detection of concurrent misuse of maps. As always, if one goroutine is writing to a map, no other goroutine should be reading or writing the map concurrently. If the runtime detects this condition, it prints a diagnosis and crashes the program. The best way to find out more about the problem is to run the program under the race detector, which will more reliably identify the race and give more detail.

Probably others @camathieu's programs are missing a couple locks too :)

**Comment by carsso** [comment]
Yes, that's exactly what we have talked about.

**Comment by speed47** [comment]
Building with go1.6 with _-race_, firing up wigo and interrupting it after 1 second already detects 13 data races indeed. I might have a closer look at those.

**Comment by camathieu** [comment]
> Probably others @camathieu's programs are missing a couple locks too :)
> @speed47 I don't think so ;)

**Comment by carsso** [comment]
Concurrent map read/write errors should be fixed since be68433.
Now the packages are built with the latest go version \o/

The other race errors are not fixed... yet.
